File: Advanced Methods of Programming/JavaClientServer/Services/src/main/java/Services/IServer.java
package Services;

import Model.Angajat;
import Model.Inscriere;
import Model.Participant;
import Model.ProbaExtins;
import java.util.List;

public interface IServer {
    void login(Angajat user, IObserver client);

    void logout(Angajat user, IObserver client);

    List<Angajat> getAllAngajati();

    List<ProbaExtins> getAllProbaExtins();

    Iterable<Participant> getAllParticipants(String idProba);

    Integer getMaxId();

    void addParticipant(Participant participant);

    void addInscriere(Inscriere inscriere);
}

File: Advanced Methods of Programming/Client-ServerCSharp/Server/Server/IServer.cs
using SharedDLL.Domain;
using SharedDLL.Utils;
using System.Collections.Generic;

namespace Server.Server
{
    public interface IServer
    {
        void AddInscriere(Inscriere i);
        //void EncreaseWith(int id, int encreaseWith);
        void Logout(IObserver observer);
        int GetMaxId();
        int FindUser(Angajat user, IObserver observer);
        List<ProbaDTO> GetAllProbeDTO();
        List<Participant> GetParticipantsByProba(int idProba);

        Proba FindProba(int idProba);

        void AddParticipant(Participant p);
    }
}
